### PR TITLE
Refactor CCImage and CCTexture2D

### PR DESCRIPTION
**Pull request by boyu0**
Refactor CCImage and CCTexture2D, make CCImage support for gray image data and add some convert function of convert image data to different texture data.
Build the libjepg, libpng and libtiff to support ios and mac, Let the ios and mac use the these library to parse the image file.

### CODE REVIEW

File: cocos2dx/platform/android/java/project.properties [hunk 1]
```diff
@@ -12,4 +12,4 @@
 
 android.library=true
 # Project target.
-target=android-10
+target=android-17
```
Comment (dumganhar) [inline cocos2dx/platform/android/java/project.properties]: Why did you change android-10 to android-17?

File: cocos2dx/platform/CCImage.h [hunk 2]
```diff
@@ -49,6 +58,14 @@ class CC_DLL Image : public Object
     
     Image();
     virtual ~Image();
+    
+    /**
+     @brief Determine how many mipmaps can we have. 
+     Its same as define but it respects namespaces
+    */
+    enum {
```
Comment (minggo) [inline cocos2dx/platform/CCImage.h:66]: use `enum class` instead
